### PR TITLE
#40 Display a particular student's notes

### PR DESCRIPTION
**Pull request by nicolesorial**
Contains work to display a student's notes in the quickview side pane.

This implementation adds a custom method to the `GetHooks.js`, because the current API call we have that gets this information is `/students/:id/notes`. This makes it difficult because we are trying to get a `note` from the `students` endpoint. I think this might be worth a discussion? I am a React noob so I don't know how okay this is? But it works! Fun!

Also worth noting: the note on the backend currently doesn't store `Date` in any capacity, and the frontend has a spot for that to show. We should either remove that from the frontend, or add that as a field in a `Note`

More details:
The only route in our API to get a student's notes is `/students/:id/notes`. We don't actually have a `/notes/:studentId` or `/notes/?studentId=:studentId` route.  If we had that, it would be a lot more straight forward. In my opinion, we have 3 options:
1. Use the implementation in this PR 
    * Pros: it's done lol
    * Cons: it likely means our API is not RESTful, because our endpoint is too custom (aka, not very `react-admin`-like)
2. Add a method to our API that allows us to get notes by the student id and refactor this implementation
    * Pros: This would likely be done using the `?` querying parameter, and means that the new method is flexible in case we need to handle other things in the future with notes. 
    * Cons: I actually don't know how to refactor the frontend to use this one off the top of my head (I Do Not Know React Oopsie)
3. Add a method to our API that allows us to get the note ids for a student and refactor this implementation
    * Pros: Allows us to use the `getMany` query we already have using the `/notes` source. Is the most `react-admin`-like
    * Cons: I'm not sure how "proper" this is from the backend perspective? like is there a standard against this? who's to say?

### CODE REVIEW

**Comment by sophiacrennan** [comment]
Awesome! Glad it works 🎉
I'd say it's a good idea to add `Date` as a field in `Note` in the backend and also add the additional routes. As for the implementation, I'm not sure what best practice is. We can talk about it in our next meeting.

**Comment by nicolesorial** [comment]
Pending the merge of PR sandboxnu/pharmd-tracker-backend#37, this should now use the `/notes/?student=:id` endpoint to get the notes

File: pharmd-app/src/components/Fields/NoteField.js [hunk 1]
```diff
@@ -1,10 +1,11 @@
 import React from "react";
-import {useGetOne} from "react-admin";
+import styled from "styled-components/macro";
+import tw from "tailwind.macro";
 import tw, { styled } from "twin.macro";
```
[the remark this answers is not in the document]
Comment (sophiacrennan) [reply pharmd-app/src/components/Fields/NoteField.js:4]: LOL I wrote that comment a few days ago but never hit submit

File: pharmd-app/src/components/Fields/NoteField.js [hunk 2]
```diff
@@ -35,7 +34,7 @@ const Title = styled.h3`
   height: auto;
 `;
 
-const Date = styled.h4`
+const DateLabel = styled.h4`
```
Comment (nicolesorial) [inline pharmd-app/src/components/Fields/NoteField.js:37]: Had to change the naming of this because I needed to use the actual `Date` constructor

**Comment by nicolesorial** [comment]
* In pharmd-app/src/components/Fields/NoteField.js, I had to change the name of the styled component `Date` to `DateLabel` because I needed to use the actual JS `Date` constructor.
* I also ran into a bug where the value for student status didn't match what we have in the backend, so I pulled all references to student status into a constant `STUDENT_STATUS` in the `apiConstants.js` file, similar to how we handle other API references

**Review by sophiacrennan — APPROVED** [review]
Yay looks good!

**Review by kev1n80 — APPROVED** [review]
Look good. I love the use of constants!!!!

**Pull request merged**